lib/block_view_profile.go

```diff
@@ -574,10 +575,13 @@ func (bav *UtxoView) _connectUpdateProfile(
 		return 0, 0, nil, RuleErrorProfileStakeMultipleSize
 	}
 	// If a username is set then it must adhere to a particular regex.
+
 	if len(txMeta.NewUsername) != 0 && !UsernameRegex.Match(txMeta.NewUsername) {
-		return 0, 0, nil, errors.Wrapf(RuleErrorInvalidUsername, "Username: %v", string(txMeta.NewUsername))
+		isAfterMetamaskBlockHeight := ENSRegex.Match(txMeta.NewUsername) && blockHeight > bav.Params.ForkHeights.UnlimitedDerivedKeysBlockHeight
+		if !isAfterMetamaskBlockHeight {
+			return 0, 0, nil, errors.Wrapf(RuleErrorInvalidUsername, "Username: %v", string(txMeta.NewUsername))
+		}
```

Comment:
```suggestion
		if blockHeight >= bav.Params.ForkHeights.UnlimitedDerivedKeysBlockHeight && !ENSRegex.Match(txMeta.NewUsername) {
			return 0, 0, nil, errors.Wrapf(RuleErrorInvalidUsername, "Username: %v", string(txMeta.NewUsername))
		}
```

Reply:
Removing the variable name, but keeping the logic. The suggested conditional doesn't throw an error if the block height is less than the fork height

Reply:
oh duh. good catch